Add root-level pnpm dev script

## What

Running `pnpm dev` from the monorepo root fails with `ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL`. The engine must also be pre-built before the app dev server works, with no guidance when it fails.

## Why

User testing (2026-04-12): first-time developer/user experience is confusing. The error messages don't tell you what to do.

## Scope

- Add a root `dev` script to `package.json` that builds the engine first, then starts the app dev server
- Something like: `"dev": "pnpm --filter quizzer-engine build && pnpm --filter coursequizzer dev"`

## Package

Root `package.json`

## Acceptance criteria

- `pnpm dev` from repo root builds the engine and starts the app dev server
- If the engine build fails, the error is visible before the app tries to start

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,6 @@
 # Changelog
 
-## 0.7.2 — 2026-04-12
+## 0.7.3 — 2026-04-12
 
 ### Developer Experience
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Changelog
 
+## 0.7.2 — 2026-04-12
+
+### Developer Experience
+
+- Add root-level `pnpm dev` script that builds the engine first, then starts the app dev server
+
 ## 0.7.1 — 2026-04-12
 
 ### Phase 2 Milestone Review Fixes

--- a/package.json
+++ b/package.json
@@ -5,6 +5,7 @@
   "scripts": {
     "test": "pnpm -r test",
     "build": "pnpm -r build",
+    "dev": "pnpm --filter quizzer-engine build && pnpm --filter coursequizzer dev",
     "format": "prettier --write .",
     "format:check": "prettier --check ."
   },

